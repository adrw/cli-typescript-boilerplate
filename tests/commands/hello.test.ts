const log = jest.fn();
console.log = log; // eslint-disable-line no-console

const { hello } = require("../../src/commands");
const { runWithAnswers, ENTER } = require("../utils");

describe("hello", () => {
  it("says hello with the right name", async () => {
    await runWithAnswers(hello, ["gabro", ENTER]).then(() => {
      expect(log).toHaveBeenCalledTimes(2);
      expect(log.mock.calls[0][0]).toMatchSnapshot();
      expect(log.mock.calls[1][0]).toMatchSnapshot();
    });
  });
});
