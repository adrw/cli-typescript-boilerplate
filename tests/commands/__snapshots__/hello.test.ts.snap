// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello says hello with the right name 1`] = `"Hello stranger!"`;

exports[`hello says hello with the right name 2`] = `"Oh, hello, gabro!"`;
